src/commands/setup/validation.ts

```diff
@@ -0,0 +1,42 @@
+import { ApiKeyNotProvided, DatadogSiteValueError } from "./errors";
+import { DatadogSite, datadogSites } from "./interface";
+import { SetupCommandInitialStateType, SetupCommandStateType } from "./setup";
+
+const isDatadogSite = (value: unknown): value is DatadogSite => {
+  return datadogSites.includes(value as DatadogSite);
+};
+
+export const validateInitialState = (
+  initialState: SetupCommandInitialStateType
+): SetupCommandStateType => {
+  if (initialState.bypassPrompts && !initialState.apiKey) {
+    throw new ApiKeyNotProvided(initialState.apiKey || "");
+  }
+  const datadogSite = initialState.datadogSite;
+  if (datadogSite === undefined) {
+    return {
+      ...initialState,
+      bypassPrompts: !!initialState.bypassPrompts,
+      datadogSite,
+    };
```

Comment:
Shouldn't this be different from return on L27? Shouldn't we put default site if `datadogSite === undefined`?

Reply:
No we should keep it as it is, it will be populated later. However I can refactor this to avoid issues :)